contract/src/main/kotlin/io/provenance/scope/loan/utility/DataValidationExtensions.kt

```diff
@@ -48,4 +49,21 @@ internal fun FigureTechDate?.isValidForSignedDate() = isSet() && this!!.value.is
 
 internal fun FigureTechUUID?.isValid() = isSet() && this!!.value.isNotBlank() && tryOrFalse { JavaUUID.fromString(value) }
 
-internal fun FigureTechMoney?.isValid() = this !== null
+internal fun ContractEnforcementContext.checksumValidation(parentDescription: String = "Input", checksum: FigureTechChecksum?) {
+    checksum.takeIf { it.isSet() }?.let { setChecksum ->
+        requireThat(
+            setChecksum.checksum.isNotBlank()  orError "$parentDescription must have a valid checksum string",
+            setChecksum.algorithm.isNotBlank() orError "$parentDescription must specify a checksum algorithm",
+        )
+    } ?: raiseError("$parentDescription's checksum is not set")
+}
+
+internal fun ContractEnforcementContext.moneyValidation(parentDescription: String = "Input's money", money: FigureTechMoney?) {
+    money.takeIf { it.isSet() }?.let { setMoney ->
+        requireThat(
+            setMoney.value.matches(Regex("^[-]?([0-9]+(?:[\\\\.][0-9]+)?|\\\\.[0-9]+)\$")) orError "$parentDescription must have a valid value",
```

Comment:
just curious: are values less than $1 supposed to come in as `0.15` or `.15`? In the latter case, the decimal wouldn't be picked up by this expression

Reply:
The former, with a leading 0. It just seems to be [the choice made](https://github.com/FigureTechnologies/metadata-asset-model/pull/15/files#diff-997a0b86d04334f7fea66fa0105f7ff49cbc206f8011a4c8fc70fd3dd3b6f9c5R47) 🤷🏽